configuration/postgres/postgres.go

```diff
@@ -0,0 +1,421 @@
+		}
+		if len(tbl) > maxIdentifierLength {
+			return m, fmt.Errorf(ErrorTooLongFieldLength+" - tableName : '%v'. max allowed field length is %v ", tbl, maxIdentifierLength)
+		}
+		m.configTable = tbl
+	} else {
+		return m, fmt.Errorf(ErrorMissingTableName)
+	}
+	// configure maxTimeout if provided
+	if mxTimeout, ok := cmetadata.Properties[connMaxIdleTimeKey]; ok && mxTimeout != "" {
+		if t, err := time.ParseDuration(mxTimeout); err == nil && t > 0 {
+			m.maxIdleTimeout = t
+		}
+	}
+	if m.maxIdleTimeout == 0 {
+		m.maxIdleTimeout = defaultMaxConnIdleTime
+	}
+	return m, nil
+}
+
+func Connect(ctx context.Context, conn string, maxTimeout time.Duration) (*pgxpool.Pool, error) {
+	config, err := pgxpool.ParseConfig(conn)
+	if err != nil {
+		return nil, fmt.Errorf("postgres configuration store connection error : %s", err)
+	}
+	pool, err := pgxpool.NewWithConfig(ctx, config)
+	if err != nil {
+		return nil, fmt.Errorf("postgres configuration store connection error : %s", err)
+	}
+	pingErr := pool.Ping(ctx)
+	if pingErr != nil {
+		return nil, fmt.Errorf("postgres configuration store ping error : %s", pingErr)
+	}
+	return pool, nil
+}
+
+func buildQuery(req *configuration.GetRequest, configTable string) (string, []interface{}, error) {
+	var query string
+	var params []interface{}
+	if len(req.Keys) == 0 {
+		query = "SELECT * FROM " + configTable
+	} else {
+		var queryBuilder strings.Builder
+		queryBuilder.WriteString("SELECT * FROM " + configTable + " WHERE KEY IN (")
+		var paramWildcard []string
+		paramPosition := 1
+		for _, v := range req.Keys {
+			paramWildcard = append(paramWildcard, "$"+strconv.Itoa(paramPosition))
+			params = append(params, v)
+			paramPosition++
+		}
+		queryBuilder.WriteString(strings.Join(paramWildcard, " , "))
+		queryBuilder.WriteString(")")
+		query = queryBuilder.String()
+
+		if len(req.Metadata) > 0 {
+			var s strings.Builder
+			i, j := len(req.Metadata), 0
+			s.WriteString(" AND ")
+			for k, v := range req.Metadata {
+				temp := "$" + strconv.Itoa(paramPosition) + " = " + "$" + strconv.Itoa(paramPosition+1)
+				s.WriteString(temp)
+				params = append(params, k, v)
+				paramPosition += 2
+				if j++; j < i {
+					s.WriteString(" AND ")
+				}
+			}
+			query += s.String()
+		}
+	}
+	return query, params, nil
+}
+
+func (p *ConfigurationStore) isSubscriptionActive(req *configuration.SubscribeRequest) (string, bool) {
+	for _, channel := range req.Metadata {
+		for key2, sub := range p.ActiveSubscriptions {
+			if res := strings.EqualFold(channel, key2); res {
+				return sub.uuid, true
+			}
+		}
+	}
+	return " ", false
+}
+
+func (p *ConfigurationStore) isSubscribed(subscriptionID string, channel string, key string) bool {
+	if val, yes := p.ActiveSubscriptions[channel]; yes {
+		if val.uuid == subscriptionID && slices.Contains(val.keys, key) {
+			return true
+		}
+	}
+	return false
+}
+
+func validateInput(keys []string) error {
+	if len(keys) == 0 {
+		return nil
+	}
```

Comment:
not needed ... as the for loop on range will not be run if the length is zero

Reply:
Add unit test for the same ....

Reply:
https://github.com/dapr/components-contrib/blob/f13fcf1ef336a1bd3821264ee04eec7985710046/configuration/postgres/postgres_test.go#L105-L114